components/datetime/src/format/datetime.rs

```diff
@@ -286,38 +324,64 @@ where
                     }
                 }
             };
-            format_number(w, value, field.length)?
+            format_number(
+                w,
+                fixed_decimal_format,
+                FixedDecimal::from(value),
+                field.length,
+            )?
         }
         FieldSymbol::Minute => format_number(
             w,
-            usize::from(
+            fixed_decimal_format,
+            FixedDecimal::from(usize::from(
```

Comment:
Nit, here and elsewhere: You should not need to convert between all these different data types, like `u8` to `usize` to `isize`. FixedDecimal supports all of them in `From` impls.